internal/servers/controller/handlers/managed_groups/managed_group_service_test.go

```diff
@@ -0,0 +1,987 @@
+	"github.com/hashicorp/boundary/internal/types/action"
+	"github.com/hashicorp/boundary/internal/types/scope"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/genproto/protobuf/field_mask"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/protobuf/testing/protocmp"
+	"google.golang.org/protobuf/types/known/structpb"
+	"google.golang.org/protobuf/types/known/timestamppb"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+)
+
+var oidcAuthorizedActions = []string{
+	action.NoOp.String(),
+	action.Read.String(),
+	action.Update.String(),
+	action.Delete.String(),
+}
+
+const testFakeManagedGroupFilter = `"/foo" == "bar"`
+
+func TestNewService(t *testing.T) {
+	conn, _ := db.TestSetup(t, "postgres")
+	rw := db.New(conn)
+	wrap := db.TestWrapper(t)
+	kmsCache := kms.TestKms(t, conn, wrap)
+	oidcRepoFn := func() (*oidc.Repository, error) {
+		return oidc.NewRepository(rw, rw, kmsCache)
+	}
+
+	cases := []struct {
+		name     string
+		oidcRepo common.OidcAuthRepoFactory
+		wantErr  bool
+	}{
+		{
+			name:    "nil-oidc-repo",
+			wantErr: true,
+		},
+		{
+			name:     "success",
+			oidcRepo: oidcRepoFn,
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			_, err := managed_groups.NewService(tc.oidcRepo)
+			if tc.wantErr {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+		})
+	}
+}
+
+func TestGet(t *testing.T) {
+	ctx := context.Background()
+	conn, _ := db.TestSetup(t, "postgres")
+	rw := db.New(conn)
+	wrap := db.TestWrapper(t)
+	kmsCache := kms.TestKms(t, conn, wrap)
+	oidcRepoFn := func() (*oidc.Repository, error) {
+		return oidc.NewRepository(rw, rw, kmsCache)
+	}
+	iamRepoFn := func() (*iam.Repository, error) {
+		return iam.NewRepository(rw, rw, kmsCache)
+	}
+
+	s, err := managed_groups.NewService(oidcRepoFn)
+	require.NoError(t, err, "Couldn't create new managed groups service.")
+
+	org, _ := iam.TestScopes(t, iam.TestRepo(t, conn, wrap))
+
+	databaseWrapper, err := kmsCache.GetWrapper(ctx, org.PublicId, kms.KeyPurposeDatabase)
+	require.NoError(t, err)
+	oidcAm := oidc.TestAuthMethod(
+		t, conn, databaseWrapper, org.PublicId, oidc.ActivePrivateState,
+		"alice-rp", "fido",
+		oidc.WithIssuer(oidc.TestConvertToUrls(t, "https://www.alice.com")[0]),
+		oidc.WithSigningAlgs(oidc.RS256),
+		oidc.WithApiUrl(oidc.TestConvertToUrls(t, "https://www.alice.com/callback")[0]),
+	)
+	omg := oidc.TestManagedGroup(t, conn, oidcAm, testFakeManagedGroupFilter)
+
+	oidcWireManagedGroup := pb.ManagedGroup{
+		Id:                omg.GetPublicId(),
+		AuthMethodId:      omg.GetAuthMethodId(),
+		CreatedTime:       omg.GetCreateTime().GetTimestamp(),
+		UpdatedTime:       omg.GetUpdateTime().GetTimestamp(),
+		Scope:             &scopepb.ScopeInfo{Id: org.GetPublicId(), Type: scope.Org.String(), ParentScopeId: scope.Global.String()},
+		Version:           1,
+		Type:              "password",
```

Comment:
Think the failing test might just be a broken fixture?

```suggestion
		Type:              "oidc",
```